src/main/java/uk/gov/justice/hmpps/prison/aop/connectionproxy/OracleConnectionAspect.kt

```diff
@@ -62,8 +64,14 @@ class OracleConnectionAspect(
     }
 
     return (this.unwrap(Connection::class.java) as OracleConnection)
-      .also { oracleConnection ->
-        oracleConnection.openProxySession(OracleConnection.PROXYTYPE_USER_NAME, info)
+      .apply {
+        try {
+          openProxySession(OracleConnection.PROXYTYPE_USER_NAME, info)
+        } catch (e: SQLException) {
+          if (e.errorCode == 1017) throw ProxyConnectionTransactionException("Could not open proxy session as $currentUsername - account doesn't support proxy connections", e)
+          if (e.errorCode == 28000) throw ProxyConnectionTransactionException("Could not open proxy session as $currentUsername - account is locked", e)
```

Comment:
this comes through as
```
Caused by: java.sql.SQLException: ORA-28000: the account is locked
```
which means that the Oracle account is locked so we can't proxy as the user